cloudbuild.yaml

```diff
@@ -0,0 +1,25 @@
+timeout: 1200s
+options:
+  volumes:
+  - name: go-modules
+    path: /go
+  env:
+  - GOPROXY=https://proxy.golang.org
+  - PROJECT_ROOT=github.com/transparency-dev/witness
+  - GOPATH=/go
+
+# Cloud Build logs sent to GCS bucket
+logsBucket: 'gs://trillian-cloudbuild-logs'
+
+steps:
+- id: 'lint'
+  name: "golangci/golangci-lint"
```

Comment:
Will this just run whatever is the latest version? We will want something hermetic or incoming PRs will be blocked on fixing up incidental changes because the ruleset has changed.

Reply:
Pinned to 1.51.

Broadly agree about random drive-by blockings, but I think it is helpful to be aware of when the codebase needs bringing into line with latest best practices (and it was certainly useful that we got the returned-errors-being-ignored warning here), but perhaps there's a better path for that which won't randomly ding PRs.

Sadly `dependabot` [doesn't currently support updating build steps in GCB](https://github.com/dependabot/dependabot-core/issues/2140).

Reply:
I'm totally supportive of following the latest version, but it should be done with a config file checked into the repo. Shame that dependabot has a blind spot here.